EFCore.InMemory.HierarchyId/Infrastructure/InMemoryHierarchyIdOptionsExtension.cs

```diff
@@ -48,8 +48,11 @@ private new InMemoryHierarchyIdOptionsExtension Extension
 
             public override bool IsDatabaseProvider => false;
 
-            public override long GetServiceProviderHashCode() => 0;
+            public override int GetServiceProviderHashCode() => 0;
 
+            public override bool ShouldUseSameServiceProvider(DbContextOptionsExtensionInfo other)
+                => true;
```

Comment:
This would be safer:
```suggestion
                => other is ExtensionInfo;
```